velox/dwio/dwrf/reader/ReaderBase.cpp

```diff
@@ -154,29 +172,53 @@ ReaderBase::ReaderBase(
     input_->load(LogType::FOOTER);
   }
 
-  auto footerStream = input_->read(
-      fileLength_ - psLength_ - footerSize - 1, footerSize, LogType::FOOTER);
+  BufferPtr remainingFooterBuffer;
+  char* footerStart;
+  if (offset >= footerSize) {
+    offset -= footerSize;
+    footerStart = buf + offset;
+  } else {
+    remainingFooterBuffer =
```

Comment:
Shall we replace footBuffer_ with remainingFooterBuffer here?

Reply:
No in this case we already running out of `footBuffer_` and anything else is read on demand now (they are also fairly large anyway at this point)

Reply:
No need to do so because at this point `fullFooterBuffer` will only contains file footer (not stripe metadata cache, and stripe metadata cache will need to be fetched separately anyway if it is needed).  Also stripe metadata cache is usually large (like 2x than the file footer along), so it's ok to fetch it separately.